scripts/loadAssets.py

```diff
@@ -2,116 +2,119 @@
+from requests.packages.urllib3.util.retry import Retry
 UTIL = Utility()
+logging, rotating_handler = UTIL.get_logging()
+logger = logging.getLogger()
+
+resources = "/usr/local/kibana-7.2.0-linux-x64/resources"
+
+def check_elasticsearch_health():
+    try:
+        url = 'http://localhost:9200/_cluster/health?wait_for_status=yellow&timeout=60s'
+        session = requests.Session()
+        response = session.get(url)
+        if not response.ok:
+            response.raise_for_status()
+
+        logger.info("elasticsearch status: " + "\n" + UTIL.pretty_format(response.json()))
+        return True
+
+    except Exception as err:
+        logger.warning("[loadAssets.py] Caught HTTP exception: {0}".format(err))
+        return False
+
+def set_default_index_pattern():
+    try:
+        url = 'http://localhost:9200/.kibana/_update/config:7.2.0'
 
-resources = "/usr/local/kibana-" + esUtil.KIBANA_VERSION + "-linux-x64/resources"
-dashboards_path = resources + "/dashboards"
-visualizations_path = resources + "/visualizations"
-searches_path = resources + "/searches"
+        postJson = {
+            "doc": {
+                "config": {
+                    "defaultIndex" : "361f5c00-b47c-11e9-86a0-cd3d7bf2f81b"
+                }
+            }
+        }
 
-dashboard_type = "dashboard"
-visualization_type = "visualization"
-search_type = "search"
+        session = requests.Session()
+        # Retry on Internal Server Error (500) and Service Unavailable (503)
+        retries = Retry(total=5, backoff_factor=0.3, status_forcelist=[ 500, 503])
+        session.mount('https://', HTTPAdapter(max_retries=retries))
+        session.mount('http://', HTTPAdapter(max_retries=retries))
+        esJson = json.dumps(postJson)
+        headers = { 'Content-type': 'application/json' }
+        response = session.post(url, headers=headers, data=esJson)
 
-always_update_whitelist = ["Capture-Dashboard.json"]
+        if not response.ok:
+            response.raise_for_status()
 
-def create_document_from_file(es_index, es_type, es_id, path_to_updated_json):
+        logger.info("Successfully set index pattern:\n" + UTIL.pretty_format(response.json()))
+        return True
+
+    except Exception as err:
+        logger.warning("[loadAssets.py] Failed to set default index pattern: {0}".format(err) \
+            + "\n\turl: {0}".format(url) + "\n\tpostJson: {0}".format(postJson))
+        return False
+
+def put_document_from_file(es_id, path_to_updated_json):
     content = UTIL.read_json_from_file(path_to_updated_json)
-    return esUtil.function_with_timeout(esUtil.STARTUP_TIMEOUT,
-                                      esUtil.create_document,
-                                          es_index,
-                                          es_type,
-                                          es_id,
-                                          content)
+    try:
+        url = 'http://localhost:9200/.kibana/_doc/' + es_id
```

Comment:
we might want to grab the id from the imported file? If ever the `es_id` and the `id` from the dashboard differ, I'm not sure which id elastic search would take